arm-appinsights/2015-05-01/swagger/component_API.json

```diff
@@ -0,0 +1,376 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "name": "InsightProperties",
+            "description": "Properties that need to be specified to create an Application Insights component.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponent"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request when creating or updating an Application Insights component. The updated component is returned.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponent"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ComponentCreate": { "$ref": "../examples/ComponentCreate.json" },
+          "ComponentUpdate": { "$ref": "../examples/ComponentUpdate.json" }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "ApplicationInsightsComponent": {
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "The kind of application that this component refers to, used to customize UI. This value is a freeform string, values should typically be one of the following: web, ios, other, store, java, phone."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationInsightsComponentProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "required": [
+        "kind"
+      ],
+      "description": "An Application Insights component definition."
+    },
+    "ApplicationInsightsComponentProperties": {
+      "description": "Properties that define an Application Insights component resource.",
+      "properties": {
+        "ApplicationId": {
```

Comment:
Swagger should correctly and completely represent the wire format for a given api-version. No exceptions to this rule. When the service removes it and does not send it on the wire then this should be removed, otherwise not.